shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -473,6 +482,65 @@ public boolean performEditorAction(int actionCode) {
     return true;
   }
 
+  @Override
+  public boolean commitContent(InputContentInfo inputContentInfo, int flags, Bundle opts) {
+    // Ensure permission is granted.
+    if (BuildCompat.isAtLeastNMR1()
+        && (flags & InputConnectionCompat.INPUT_CONTENT_GRANT_READ_URI_PERMISSION) != 0) {
+      try {
+        inputContentInfo.requestPermission();
```

Comment:
Does this request show a dialog for permissions? If so, I would recommend keeping all UI-related behavior out of this class. Consider making it the clients responsibility to obtain the appropriate permissions before running the code that requires the permissions. You can throw an exception if that permission isn't granted.

Reply:
It does not appear to show a dialog for any permissions. I think its more of asking the system to have temporary access to the committed files, if the app isn't given access already.

https://developer.android.com/reference/android/view/inputmethod/InputContentInfo#requestPermission()

Reply:
Have you tried running the code without the permissions to see what happens? If no dialog is shown, then who are you "requesting permission" from? Typically a "request" is something that can be denied, so this seems strange.